### PR TITLE
Fix: Bug Affecting Widgets without description

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
## Description
Fixes: https://github.com/WordPress/gutenberg/issues/14552

We were not checking in the widget description was set and on widgets, without description, an undefined notice was being thrown.

## How has this been tested?
I installed the AnsPress Question Answer plugin: https://wordpress.org/plugins/anspress-question-answer.
I created a new page and verified and checked Undefined notice was not being thrown in the PHP code. In master, a notice should appear.

### CODE REVIEW

File: lib/widgets.php [hunk 1]
```diff
@@ -110,7 +115,7 @@ function gutenberg_legacy_widget_settings( $settings ) {
 		}
 		$available_legacy_widgets[ $widget_id ] = array(
 			'name'             => html_entity_decode( $widget_obj['name'] ),
-			'description'      => null,
+			'description'      => html_entity_decode( wp_widget_description( $widget_id ) ),
```
Comment (aduth) [inline lib/widgets.php]: In what way is this change related to the fix?

**Review by aduth — COMMENTED** [review]
Alternative test (I'd created before this pull request):

1. Ensure `define( 'WP_DEBUG', true );` in `wp-config.php`
2. Add the following as `wp-content/mu-plugins/custom-widget.php`:

```php
<?php

class Custom_Widget extends WP_Widget {
	public function __construct() {
		parent::__construct( 'Custom_Widget', 'Custom Widget' );
	}

	public function widget( $args, $instance ) {
	}

	public function form( $instance ) {
	}

	public function update( $new_instance, $old_instance ) {
		return $new_instance;
	}
}

add_action( 'init', function() {
	register_widget( 'Custom_Widget' );
} );
```

3. Navigate to Posts > Add New

The PHP warning notice is no longer shown after these changes.

Can you confirm: Prior to this change, `description` was passed as an empty string when not defined by the widget. It is now passed as `null`. Is there a meaningful impact of this difference?

**Comment by jorgefilipecosta** [comment]
Hi @aduth,
In my tests, there is no difference between returning a null or empty string. In legacy widgets description is only used in the inspector, and empty string and null are both unrendered in React.
https://github.com/WordPress/gutenberg/blob/b421fc9574a7f437fb5068f8b6100db8f2926956/packages/block-library/src/legacy-widget/edit.js#L95-L97

File: lib/widgets.php (hunk 1 above)
Comment (jorgefilipecosta) [reply lib/widgets.php]: During these changes, I noticed I missed to pass a description when I added support some simple call back widgets.
But you are right it is not directly related to these changes I separated this change to other PR https://github.com/WordPress/gutenberg/pull/14615.

**Review by aduth — APPROVED** [review]
